TumbleBitSetup/RsaKey.cs

```diff
@@ -68,6 +79,18 @@ internal byte[] Decrypt(byte[] encrypted)
             return engine.ProcessBlock(encrypted, 0, encrypted.Length);
         }
 
+        public PoupardSternProof CreatePoupardSternProof(PoupardSternSetup setup)
```

Comment:
The cryptogrpaher oriented classes that should be easy to copy paste to BC are the two setup classes, and the `PoupardSternTest.cs`, `PermutationTest.cs`. The rest is dev oriented layer.